lib/signer.ex

```diff
@@ -62,7 +62,12 @@ defmodule AntikytheraAws.Signer do
           ad = Time.to_iso_basic(Time.now())
           {ad, Map.put(without_amz_date, "x-amz-date", ad)}
       end
-      sign_ready_headers = if st, do: Map.put(headers_with_date, "x-amz-security-token", st), else: headers_with_date
+      headers_might_with_st = if st, do: Map.put(headers_with_date, "x-amz-security-token", st), else: headers_with_date
+      sign_ready_headers    = if service == "s3" and method in [:post, :put] do
+        Map.put(headers_might_with_st, "x-amz-content-sha256", hex_sha256(payload))
```

Comment:
`payload` here can be a very large binary, so calculating its `hex_sha256` twice is not an option.
Pre-compute the value and pass it to `canonical_request/6`.